### PR TITLE
Light client improvements

### PR DESCRIPTION
**Pull request by rphmeier**
- a few bug fixes in `on_demand` and `parity_rpc`
- blockchain import command for light clients.

### CODE REVIEW

**Comment by rphmeier** [comment]
@arkpar d42d798 and 3ee5390 may be worth backporting before 1.7

**Comment by arkpar** [comment]
Contains commits from https://github.com/paritytech/parity/pull/6148, probably needs a rebase

File: ethcore/light/src/on_demand/mod.rs [hunk 1]
```diff
@@ -54,13 +54,13 @@ struct Peer {
...
 impl Peer {
-	// whether this peer can fulfill the
+	// whether this peer can fulfill the necessary capabilities
 	fn can_fulfill(&self, c: &Capabilities) -> bool {
 		let caps = &self.capabilities;
 
-		caps.serve_headers == c.serve_headers &&
-			caps.serve_chain_since >= c.serve_chain_since &&
-			caps.serve_state_since >= c.serve_chain_since
+		caps.serve_headers >= c.serve_headers &&
```
Comment (arkpar) [inline ethcore/light/src/on_demand/mod.rs]: `serve_headers` is a bool as far as I can see. Why `>=` ?

File: ethcore/light/src/on_demand/mod.rs [hunk 2]
```diff
@@ -54,13 +54,13 @@ struct Peer {
...
+			(c.serve_chain_since.is_none() || caps.serve_chain_since <= c.serve_chain_since) &&
+			(c.serve_state_since.is_none() || caps.serve_state_since <= c.serve_state_since)
```
Comment (arkpar) [inline ethcore/light/src/on_demand/mod.rs]: Why does `is_none` allow a request?
Shouldn't  chain and state numbers be strictly lower?

File: ethcore/light/src/on_demand/mod.rs (hunk 1 above)
Comment (rphmeier) [inline ethcore/light/src/on_demand/mod.rs]: the purpose of this function is to check whether the capabilities `c` are encompassed by the capabilities `caps`. If `c.serve_headers` is `false`, `caps.serve_headers` can be `false` or `true. If `c.serve_headers` is `true`, `caps.serve_headers` can only be `true`.

File: ethcore/light/src/on_demand/mod.rs (hunk 2 above)
Comment (rphmeier) [inline ethcore/light/src/on_demand/mod.rs]: should probably be rewritten as `c.serve_chain_since >= caps.serve_chain_since`

Comment (rphmeier) [inline ethcore/light/src/on_demand/mod.rs]: if the request doesn't require any state or chain serving, then it doesn't matter what the peer capabilities are. If the request does require chain or state serving, then the block number this peer supports beyond should be `<=` the request required block number. but this should also guard against `caps.serve_since.is_none()`. I will rewrite for clarity

**Pull request merged**